tests/assets/dashAgGridComponentFunctions.js

```diff
@@ -113,4 +113,28 @@ dagcomponentfuncs.myCustomButton = function (props) {
 dagcomponentfuncs.myCustomTooltip = function (params) {
     info = [React.createElement('div',{},params.data.ticker), React.createElement('div',{}, params.data.company),React.createElement('div',{}, params.data.price)]
     return React.createElement('div',{style: {"border":'2pt solid white', 'backgroundColor':'black'}}, info);
+}
+
+dagcomponentfuncs.myCustomButton2 = function (props) {
+
+    const {setProps, data} = props;
+
+    if (!props.value) {
+        return React.createElement('button')
+    }
+
+    function onClick() {
+        setProps({cellRendererData: 'updated'})
+    }
+
+    const id = JSON.stringify({'index': props.rowIndex, 'type':'customButton'})
```

Comment:
Might be a performance issue with lots of components?

Reply:
It’s just really an example of an id, the id isn’t actually necessary, it just helps react to stop complaining. 

Or were you talking about the setProps?